### PR TITLE
[SPARK-12159] [ML] Add user guide section for IndexToString transformer

### PR DESCRIPTION
**Pull request by BenFradet**
Documentation regarding the `IndexToString` label transformer with code snippets in Scala/Java/Python.

### CODE REVIEW

**Comment by holdenk** [comment]
It might be useful to also document the different ways "missing" labels can be handled - what are your thoughts?

**Comment by BenFradet** [comment]
Hey @holdenk, thanks for reviewing.

Do you mean regarding StringIndexer#setHandleInvalid method? If so, yes that'd be a good addition.

However, I'm not sure if I should include it in this jira/pr or create another, input welcome.

**Comment by BenFradet** [comment]
cc @jkbradley

**Comment by holdenk** [comment]
That is what I was referring to, handling it in a follow up JIRA/PR seems ok too (just since one of the things blocking the original implementation was wanting to have it be user controllable if we allowed people to specify their own maps it seemed like good for that to also make it through to the docs).

**Comment by jkbradley** [comment]
Thanks for the PR; I'll take a look now!
@holdenk handleInvalid should be in a separate PR since it's for StringIndexer, but I agree it'd be nice to add to the docs.

File: docs/ml-features.md [hunk 1]
```diff
@@ -951,9 +951,157 @@ indexed.show()
...
+~~~~
+
+<div class="codetabs">
+<div data-lang="scala" markdown="1">
+
+Refer to the [IndexToString Scala docs](api/scala/index.html#org.apache.spark.ml.feature.IndexToString)
+for more details on the API.
+
+{% highlight scala %}
+import org.apache.spark.ml.feature.{IndexToString, StringIndexer}
```
Comment (jkbradley) [inline docs/ml-features.md]: Would you mind moving these to examples/ and pulling the code snippets into here using the include_example functionality?  You can find examples of include_example in this .md file.  This makes the examples easier to test & maintain.

File: docs/ml-features.md [hunk 2]
```diff
@@ -951,9 +951,157 @@ indexed.show()
...
+val indexer = new StringIndexer()
+  .setInputCol("category")
+  .setOutputCol("categoryIndex")
+  .fit(df)
+val indexed = indexer.transform(df)
+
+val converter = new IndexToString()
+  .setInputCol("categoryIndex")
+  .setOutputCol("originalCategory")
+  .setLabels(indexer.labels)
```
Comment (jkbradley) [inline docs/ml-features.md]: You probably don't need to specify labels; they should be pulled from column metadata.

**Comment by jkbradley** [comment]
Those are my only comments; the examples look good.  Btw, it's OK this time, but in general, I'd recommend doing little cleanups in a separate PR.  Especially when lots of docs are being merged, it's really easy to hit merge conflicts.  Thanks!  I'll watch for updates.

**Comment by BenFradet** [comment]
@jkbradley Thanks for reviewing, will take those comments into account.

**Comment by jkbradley** [comment]
That was a spurious test failure; I asked it to retest

**Comment by jkbradley** [comment]
```dev/lint-python``` should catch these issues

**Comment by jkbradley** [comment]
LGTM except for the Python style issue

**Comment by jkbradley** [comment]
Merging with master and branch-1.6

Thanks for the PR!

**Comment by BenFradet** [comment]
@jkbradley Should I log a jira for completing the user guide on StringIndexer regarding the handling of missing labels @holdenk was talking about?

**Comment by jkbradley** [comment]
If you wouldn't mind, that'd be great, thanks!

**Comment by BenFradet** [comment]
will do

**Pull request closed without merging**